app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -206,9 +206,22 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
     override fun onStop() {
         super.onStop()
         updateLastBrowseActivity()
+        if (FeatureFlags.historyMetadataFeature) {
+            updateHistoryMetadata()
+        }
         pwaOnboardingObserver?.stop()
     }
 
+    private fun updateHistoryMetadata() {
+        getCurrentTab()?.let { tab ->
+            if (tab is TabSessionState) {
```

Comment:
nit: could be `as?` to avoid extra nesting

Reply:
👍 updated.